### DISCUSSION TITLE
Add a threshold price and mode to target entities

### DISCUSSION THREAD

**Discussion by d1964b** [BottlecapDave/HomeAssistant-OctopusEnergy#846] [Ideas]
Cheap prices don't come around that often these days so it would be great to be able to easily take advantage of them when they do. I've been thinking it would be neat to add the notion of a threshold price and a 'mode' to your target entities. Such as  Ignore, Merge, Extend Merge, Threshold and a seperate numerical value of Threshold. These would act as follows: 
a - 00:00-00:30  0.0p
b - 00:30-01:00  -0.1p
c - 01:00-01:30  1.0p
d - 01:30-02:00  0.0p
e - 02:00-02:30  1.0p
f - 02:30-03:00  1.0p
g - 03:00-03:30  0.0p

Threshold : As per example

Request: cheapest 1hr between 00:30 and 03:00

Ignore         : Act as now (i.e. you will definetly get your request )
Merge          : Act as now , plus any values inside the time range of request that are less than or equal to threshold
Extended Merge : Act as now , plus any value outside/inside your request time window which is lte your threshold
Threshold      : Ignore time range, no guarantee request will be met , but simply be driven by threshold  

giving results like this : 
Ignore        : b and d
Merge         : If threshold is 0 , then select cheapest values as per request [b, d] + values lte threshold [b,d] = [b, d]
Merge         : If threshold is -0.01 , then select cheapest values as per request [b, d] + values lte threshold [b] = [b, d]
Merge         : If threshold is 1.0p , then select cheapest values as per request [b, d] + values lte threshold [b,c,d,e,f] = [b,c,d,e,f]
Extend Merge  : If threshold is 0 , then select cheapest values as per request [b, d] + values lte threshold [a,b,d,g] = [a,b,d,g]
Plunge        : If threshold is 0 , then select [a,b,d,g]

Really enjoyng your integration , thanks for all the work you have put into it.

**Comment by BottlecapDave** [comment]
Hello and sorry for the late response to this.

It feels like you've potentially over complicated the purpose of the target rate sensor? In your examples it feels like it would be best to just look at the target rate sensor and compare with your thresholds to determine if a device should turn on/off? Unless you're doing something else with the target rate sensor?
